src/elements/unified-share-modal-element/UnifiedShareModalElement.js

```diff
@@ -0,0 +1,172 @@
+
+    const getError = useCallback(() => {
+        setErrorExists(true);
+    }, [setErrorExists]);
+
+    const getUserSuccess = useCallback(
+        userData => {
+            const { id, userEnterpriseData } = normalizeUserResponse(userData);
+            setSharedLink({ ...sharedLink, ...userEnterpriseData });
+            setCurrentUserID(id);
+            setErrorExists(false);
+        },
+        [sharedLink],
+    );
+    const getItemSuccess = (itemData: USMElementItemAPIResponse) => {
+        const { item: itemFromAPI, sharedLink: sharedLinkFromAPI } = normalizeItemResponse(itemData);
+        setItem(itemFromAPI);
+        setSharedLink(sharedLinkFromAPI);
+        setErrorExists(false);
+    };
+
+    useEffect(() => {
+        resetState();
+    }, [resetState, token, itemID, itemType]);
+
+    useEffect(() => {
+        const getUserData = async () => {
+            await api.getUsersAPI(false).getUser(itemID, getUserSuccess, getError, {
+                fields: [FIELD_ENTERPRISE, FIELD_HOSTNAME],
+            });
+        };
+        if (item && sharedLink && !currentUserID) {
+            getUserData();
+        }
+    });
+
+    useEffect(() => {
+        const getItem = async () => {
+            if (itemType === TYPE_FILE) {
+                await api.getFileAPI().getFile(itemID, getItemSuccess, getError, {
+                    fields: [
+                        FIELD_DESCRIPTION,
+                        FIELD_EXTENSION,
+                        FIELD_ID,
+                        FIELD_NAME,
+                        FIELD_PERMISSIONS,
+                        FIELD_SHARED_LINK,
+                        FIELD_SHARED_LINK_FEATURES,
+                        FIELD_TYPE,
+                    ],
+                });
+            }
+
+            if (itemType === TYPE_FOLDER) {
+                await api.getFolderAPI().getFolderFields(itemID, getItemSuccess, getError, {
+                    fields: [
+                        FIELD_DESCRIPTION,
+                        FIELD_EXTENSION,
+                        FIELD_ID,
+                        FIELD_NAME,
+                        FIELD_PERMISSIONS,
+                        FIELD_SHARED_LINK,
+                        FIELD_SHARED_LINK_FEATURES,
+                        FIELD_TYPE,
+                    ],
+                });
+            }
+        };
+
+        if (!item && !sharedLink) {
+            getItem();
+        }
+    }, [api, getError, item, itemID, itemType, sharedLink, currentUserID]);
+
+    const renderElement = () => {
+        if (errorExists) {
+            return <ErrorMask errorHeader={<FormattedMessage {...elementMessages.loadingError} />} />;
+        }
+
+        if (item && sharedLink) {
+            return (
+                <Internationalize language={language} messages={messages}>
+                    <UnifiedShareModal
+                        canInvite={sharedLink.canInvite}
+                        collaboratorsList={{ collaborators: [] }}
```

Comment:
Can this be derived from having access to the item and an API instance, or do we have to provide it via an external prop?

Reply:
Yes, I'll request this data from the Collaborators API in a follow-up PR. I added a to-do note next to this line.